.github/workflows/pr-one-shot-test.yml

```diff
@@ -0,0 +1,36 @@
+---
+name: PR One-Shot Test
+
+on:
+  pull_request_review:
+    types: ["submitted"]
+
+jobs:
+  one-shot-test:
+    # Only run when a pull request is approved
+    if: github.event.review.state == 'approved'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Set up Python 3.13
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.13'
+
+      - name: Install uv
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          echo "$HOME/.cargo/bin" >> $GITHUB_PATH
+
+      - name: Run one-shot test
+        env:
```

Comment:
there's a lot more than just this one variable, in the event we're using different base url